source/widgets/ProjectStatus/js/configuration.js

```diff
@@ -29,13 +30,25 @@ function OctopusStatusWidgetConfiguration() {
                 return customSettings;
             };
 
-            var fetchDataSourceContent = function (queryUri, token, name, spaceId) {
-                let query;
+            var fetchDataSourceContent = function (queryUri, token, name, spaceId, partialNameFilter) {
+                let parameters = [];
                 if (spaceId) {
-                    query = '{"dataSourceDetails": {"dataSourceName":"' + name + '", "parameters": {"SpaceId":"' + spaceId + '"} } }';
+                    parameters.push('"SpaceId":"' + spaceId + '"');
+                }
+
+                // We check explicitly for `null` in this case. There are cases where we need to pass an empty string through to our endpointUrl.
```

Comment:
You mean `null` and `""` are different? :surprised_pikachu:

Reply:
Oh yea, this was a fun one ... without this it was generating this network call: `projects?partialName=null` which made it go boom. I'm not familiar enough with these parameters and how they pass through as to whether we can be more intelligent here.